### PR TITLE
Resolved pytest errors by correctly annotating ORM relationships with Mapped type

### LINKED ISSUE #88: Update SQLAlchemy > 2.0

### PR DESCRIPTION
**Pull request by roshnaeem**
# Pull Request

## Description
Fixed pytests failing because of sqlalchemy, see also https://github.com/openclimatefix/pv-site-datamodel/pull/107 

Fixes #88

### CODE REVIEW

**Comment by roshnaeem** [comment]
@peterdudfield , this PR will close #88

**Comment by roshnaeem** [comment]
@peterdudfield , i think we need to update sqlalchemy version here https://github.com/openclimatefix/pv-site-datamodel/blob/main/poetry.lock#L16 to enable it on pv-site-app.

File: pvsite_datamodel/sqlmodels.py [hunk 1]
```diff
@@ -58,11 +60,11 @@ class SiteGroupSQL(Base, CreatedMixin):
 
     # Relationships
     # N-N
-    sites: List["SiteSQL"] = relationship(
+    sites: Mapped["SiteSQL"] = relationship(
```
Comment (peterdudfield) [inline pvsite_datamodel/sqlmodels.py:63]: Im worried this is no longer a list?

Comment (roshnaeem) [reply pvsite_datamodel/sqlmodels.py:63]: According to my search, the use of `Mapped` for annotating values doesn't change its list-like behaviour at runtime; it still supports typical list operations such as appending and iteration. This change primarily enhances type clarity and tool support, aligning with SQLAlchemy's recommendations for explicit ORM mappings in version 1.4 and beyond.

**Comment by peterdudfield** [comment]
Looks like we still get tests failing - https://github.com/openclimatefix/pv-site-datamodel/actions/runs/8350094672/job/22855747461

**Comment by peterdudfield** [comment]
can you run pytest on this repo?

**Comment by roshnaeem** [comment]
@peterdudfield, I checked `pytest` before opening PR, for me all tests are passing, i have attached a screenshot here.  Can you please send me a screenshot of the error, so i can fix it?  
Screenshot 2024-03-19 at 11 15 57 PM

**Comment by peterdudfield** [comment]
Screenshot 2024-03-20 at 07 46 45
werid -

**Comment by peterdudfield** [comment]
see https://github.com/openclimatefix/pv-site-datamodel/pull/110

**Comment by roshnaeem** [comment]
@peterdudfield , let me check.

**Comment by peterdudfield** [comment]
If you make a new PR, the tests should automatically run, so you'll be able to see whats going on

**Pull request merged**